cli/src/commands/squash.rs

```diff
@@ -155,21 +217,80 @@ pub(crate) fn cmd_squash(
             ));
         }
         sources = vec![source];
-        destination = parents.pop().unwrap();
-    }
+        pre_existing_destination = Some(parents.pop().unwrap());
+    };
 
-    let matcher = workspace_command
+    workspace_command.check_rewritable(sources.iter().chain(&pre_existing_destination).ids())?;
+
+    // prepare the tx description before possibly rebasing the source commits
+    let source_ids: Vec<_> = sources.iter().ids().collect();
+    let tx_description = if let Some(destination) = &pre_existing_destination {
+        format!("squash commits into {}", destination.id().hex())
+    } else {
+        match &source_ids[..] {
+            [] => format!("squash {} commits", source_ids.len()),
```

Comment:
Can you add test for empty source revision and tree?

I don't have strong feeling whether new commit should be inserted in that case, but the result and status output should be reasonable.